src/java.naming/share/classes/com/sun/jndi/ldap/AbstractLdapNamingEnumeration.java

```diff
@@ -126,45 +170,50 @@ public final boolean hasMoreElements() {
      * Retrieve the next set of entries and/or referrals.
      */
     private void getNextBatch() throws NamingException {
+        try {
+            LdapResult newRes = homeCtx().getSearchReply(enumCtx.enumClnt, enumCtx.res);
+            enumCtx.setRes(newRes);
+            if (enumCtx.res == null) {
```

Comment:
This looks odd, setting the value using synchronized, but reading it without.

Reply:
I've added getters to EnumCtx, and a comment explaining why only setters are synchronized.